### PR TITLE
added data efficiency experiments on segmentation

### PR DESCRIPTION
**Pull request by h-spiess**
this adds the code that enables training on a subset of the training data and adds two default configs for the flat and hp models

### CODE REVIEW

**Comment by h-spiess** [comment]
in addition, it fixed the default flat configs

File: heal_swin/run_configs/segmentation/data_efficiency_swin_hp_synwoodscape_large_plus_AD_train_run_config.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+# fmt: on
+
+import os  # noqa: E402
+import subprocess  # noqa: E402
+from pathlib import Path  # noqa: E402
+
+import argparse
+import math
+
+def get_argparser_for_config_args():
```
Comment (JanEGerken) [inline heal_swin/run_configs/segmentation/data_efficiency_swin_hp_synwoodscape_large_plus_AD_train_run_config.py:16]: This goes a bit against our philosophy of only using config files and staying away from argparse as much as possible. Could you refactor this to be used as an array job, similarly to our evaluation config files? Then, the range of fractions and the seed are also documented here. For array jobs, you also have to change the naming of the slurm file in line 4 and the `job_id` should contain the task number in line 43.

File: heal_swin/run_configs/segmentation/data_efficiency_swin_synwoodscape_large_plus_AD_train_run_config.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+# fmt: on
+
+import os  # noqa: E402
+import subprocess  # noqa: E402
+from pathlib import Path  # noqa: E402
+
+import argparse
+import math
+
+def get_argparser_for_config_args():
```
Comment (JanEGerken) [inline heal_swin/run_configs/segmentation/data_efficiency_swin_synwoodscape_large_plus_AD_train_run_config.py:16]: See above

**Review by JanEGerken — CHANGES_REQUESTED** [review]
Good work, thanks Heiner! Could you remove argparse from the config files? I left a comment there. Also, could you add a few lines describing the new config files in `heal_swin/README.md` in the "Reproducing runs from the paper" section under "segmentation"? Apart from that, please run black and flake8 and we're good to go.

**Comment by h-spiess** [comment]
i think i have addressed these issues with the new commit

**Review by JanEGerken — CHANGES_REQUESTED** [review]
Great, thanks for updating the readme and the run configs!
I think you ran black with the default line length instead of the line length 100 which we use in this repo. Could you rerun black with the longer line length, e.g. by using `format-code` from `run.py`? Please also check the diffs, sometimes, black introduces commas when it shrinks lines which prevent them from being enlarged later.

**Comment by h-spiess** [comment]
okay. i've changed by black settings to not have that issue in the future and i've run `format-code`.

**Review by JanEGerken — APPROVED** [review]
Perfect, thanks!
